### PR TITLE
fix: consider blocks being processed as known

### PR DESCRIPTION
**Pull request by matklad**
closes https://github.com/near/nearcore/issues/7371

Test Plan:

Modify `test_block_sync` to explicitly hit the block-in-processing case.

### CODE REVIEW

File: chain/client/src/test_utils.rs [hunk 1]
```diff
@@ -1344,6 +1347,42 @@ impl TestEnv {
...
+    /// thread which applies the chunks on the block will get blocked until
+    /// `resume_block_processing` is called.
+    ///
+    /// Note that you must call `resume_block_processing` at some later point to
+    /// unstuck the block.
+    ///
+    /// Implementation is rather crude and just hijacks our logging
+    /// infrastructure. Hopefully this is good enough, but, if it isn't, we can
+    /// add something more robust.
+    pub fn pause_block_processing(&mut self, capture: &mut TracingCapture, block: &CryptoHash) {
```
Comment (matklad) [inline chain/client/src/test_utils.rs:1360]: @mzhangmzz especially curious if you think there's a better way to achieve this! This is the second test where we need to coax async block processing to (not) happen at a particular time.

Comment (mzhangmzz) [reply chain/client/src/test_utils.rs:1360]: Maybe we can modify the existing `wait_for_all_blocks` and `wait_for_block` methods in the BlocksInProcessing struct? Current tests call these methods to finish the block processing for all blocks. We can add another field `paused_blocks` there and exclude the block in `paused_blocks` when we call these wait for blocks methods.

Comment (matklad) [reply chain/client/src/test_utils.rs:1360]: :thinking:, I think that won't work: this'll prevent us from waiting for those blocks, but it won't prevent the blocks from actually getting done. This line is the key: 

https://github.com/near/nearcore/blob/7efd76cd6381f59e4589a6b0fd6f6c3d65492e03/chain/chain/src/chain.rs#L1579

Here, we'll process ready blocks, and we esentially want to force some blocks to stay in that channel. We *could* do that, but that'll I think require modification of production code, and not only the testonly stuff.

Comment (mzhangmzz) [reply chain/client/src/test_utils.rs:1360]: You are totally right! Sorry I didn't think very carefully.

Comment (mzhangmzz) [reply chain/client/src/test_utils.rs:1360]: I can't think of a better way. Another way I can think of is add yet another callback before we call `sc.send(block_hash.clone(), res))` to intercept the process there, but then that adds yet another callback and we already have two callback methods in the `schedule_apply_chunks` code.

**Review by mzhangmzz — APPROVED** [review]

**Pull request merged**